nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -135,29 +89,17 @@
             .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
             .build();
 
-    protected static final PropertyDescriptor SMTP_MAXIMUM_MSG_SIZE = new PropertyDescriptor.Builder()
+    static final PropertyDescriptor SMTP_MAXIMUM_MSG_SIZE = new PropertyDescriptor.Builder()
```

Comment:
Yes it goes straight to disc via the Function that is passed to the SmtpConsumer. 
As far as enforcing data size we can address it in the future if that is a problem. For example we don't have the same enforcement in PutFile and other processors unless memory is involved which in this case it isn't. 
Also, in all honesty I can't see someone using ListenSMTP or email in general to send large data or to move data. There are better more suitable protocols for it.